pkg/gitutils/gitutils_test.go

```diff
@@ -70,15 +71,33 @@ func TestCloneArgsStripFragment(t *testing.T) {
 	}
 }
 
+func gitGetConfig(name string) (string, error) {
+	b, err := git([]string{"config", "--get", name}...)
+	if err != nil {
+		return "", err
+	}
+	return strings.Trim(string(b), "\r\n"), err
+}
+
 func TestCheckoutGit(t *testing.T) {
 	root, err := ioutil.TempDir("", "docker-build-git-checkout")
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer os.RemoveAll(root)
 
+	autocrlf, err := gitGetConfig("core.autocrlf")
+	if err != nil {
```

Comment:
this `err` check isn't needed, nor it's needed to return an error from `gitGetConfig` since you're just interested in empty or non empty string. You can ignore the error in `gitGetConfig` and just return either `""` or `true|false` and use it here directly w/o this check.